ast/core/src/main/kotlin/me/tatarka/kotlin/ast/Ast.kt

```diff
@@ -195,12 +195,16 @@ abstract class AstType : AstElement() {
 
     abstract fun isPlatform(): Boolean
 
+    abstract fun isSamInterface(): Boolean
+
     abstract fun isFunction(): Boolean
 
     abstract fun isTypeAlias(): Boolean
 
     abstract fun resolvedType(): AstType
 
+    abstract fun samReturnType(): AstType
+
```

Comment:
I think it would make the impl less awkward if we had

```
abstract fun samFunction(): AstFunction?
```

Reply:
I cleaned it up a little bit. I'm less sure about removing `isSamInterface` since it's a bit clearer when calling than checking if the result of `samFunction` is null.